x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -217,62 +224,91 @@ function getDuration(items: IWaterfallItem[]) {
   return timestampEnd - timestampStart;
 }
 
+function getWaterfallItemTransactionById(
+  itemsById: IWaterfallIndex,
+  id?: IWaterfallItem['id']
+): IWaterfallItemTransaction | void {
+  if (!id) {
+    return undefined;
+  }
+
+  const item = itemsById[id];
+  if (item.docType === 'transaction') {
+    return item;
+  }
+}
+
 function createGetTransactionById(itemsById: IWaterfallIndex) {
   return (id?: IWaterfallItem['id']) => {
-    if (!id) {
-      return;
+    const waterfallItemTransaction = getWaterfallItemTransactionById(
+      itemsById,
+      id
+    );
+    if (waterfallItemTransaction) {
+      return waterfallItemTransaction.transaction;
     }
+  };
+}
 
-    const item = itemsById[id];
-    if (item.docType === 'transaction') {
-      return item.transaction;
+function createGetErrorCountByTransactionId(itemsById: IWaterfallIndex) {
+  return (id?: IWaterfallItem['id']) => {
+    const waterfallItemTransaction = getWaterfallItemTransactionById(
+      itemsById,
+      id
+    );
+    if (waterfallItemTransaction) {
+      return waterfallItemTransaction.errorCount;
```

Comment:
A few comments about `getWaterfallItemTransactionById`, `createGetTransactionById` and `createGetErrorCountByTransactionId`:

I don't think we need `createGetErrorCountByTransactionId` since we already have `errorsPerTransaction` which is a `StringMap` like `itemsById`. We can just return it as `errorCountByTransactionId` or `errorsPerTransaction` - whatever makes the naming consistent.